src/platform/forms-system/src/js/utilities/file/checkForEncryptedPdf.js

```diff
@@ -0,0 +1,39 @@
+// requestLockedPdfPassword = feature flag
+// getEncryptedPassword = file option set in uiSchema
+export default function checkForEncryptedPdf(
+  file,
+  requestLockedPdfPassword,
+  uiSchema = {},
+) {
+  const { getEncryptedPassword } = uiSchema['ui:options'] || {};
+  return new Promise((resolve, reject) => {
+    if (requestLockedPdfPassword && getEncryptedPassword) {
+      const reader = new FileReader();
+      reader.onloadend = event => {
+        if (event.target.readyState === FileReader.DONE) {
+          // const { result } = event.target // with readAsBinaryString
+          const result = Array.from(new Int8Array(event.target.result))
+            .map(v => String.fromCharCode(v))
+            .join('');
+          if (result.includes('/Encrypt')) {
+            resolve(true);
+            return;
+          }
+        }
+        resolve(false);
+      };
+      try {
+        // "/Encrypt" is within the first 200 bytes of the PDF
+        const blob = file.slice(0, 200);
+        // TODO: once we stop supporting IE11, update this and replace the
+        // readAsArrayBuffer and Int8Array conversion in the code with a string
+        // compare using readAsBinaryString (which isn't supported by IE11)
+        reader.readAsArrayBuffer(blob);
+      } catch (error) {
+        reject(error);
+      }
```

Comment:
What has the possibility of throwing the exception here? Is it `file` being null, or can the reader throw an exception as it is reading? I ask because I'm wondering if we could get this function to resolve with a consistent true/false, rather than reject with this error object.

Reply:
I just tried uploading a 1 byte text file renamed with a PDF extension, and it didn't have an error.. so I guess we can remove the try/catch block.